drivers/sensor/tdk/icm456xx/icm456xx.c

```diff
@@ -0,0 +1,1061 @@
+	} else if (val < 4 && val >= 2) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_2;
+	} else if (val < 5 && val >= 4) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_4;
+	} else if (val < 7 && val >= 5) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_5;
+	} else if (val < 8 && val >= 7) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_7;
+	} else if (val < 10 && val >= 8) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_8;
+	} else if (val < 11 && val >= 10) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_10;
+	} else if (val < 16 && val >= 11) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_11;
+	} else if (val < 18 && val >= 16) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_16;
+	} else if (val < 20 && val >= 18) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_18;
+	} else if (val < 32 && val >= 20) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_20;
+	} else if (val < 64 && val >= 32) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_32;
+	} else if (val ==62) {
+		avg_enum = IPREG_SYS2_REG_129_ACCEL_LP_AVG_64;
+	}
+	return avg_enum;
+}
+
+static uint8_t convert_enum_to_acc_fs(uint8_t fs_enum)
+{
+        uint8_t acc_fs = 0;
+
+        if (fs_enum == ACCEL_CONFIG0_ACCEL_UI_FS_SEL_2_G) {
+                acc_fs = 2;
+        } else if (fs_enum == ACCEL_CONFIG0_ACCEL_UI_FS_SEL_4_G) {
+                acc_fs = 4;
+        } else if (fs_enum == ACCEL_CONFIG0_ACCEL_UI_FS_SEL_8_G) {
+                acc_fs = 8;
+        } else if (fs_enum == ACCEL_CONFIG0_ACCEL_UI_FS_SEL_16_G) {
+                acc_fs = 16;
+        }
+        return acc_fs;
+}
+
+uint16_t convert_enum_to_gyr_fs(uint8_t fs_enum)
+{
+        uint16_t gyr_fs = 0;
+
+	if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_15_625_DPS) {
+		gyr_fs = 15;
+	} else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_31_25_DPS) {
+		gyr_fs = 31;
+	} else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_62_5_DPS) {
+		gyr_fs = 62;
+	} else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_125_DPS) {
+		gyr_fs = 125;
+	} else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_250_DPS) {
+                gyr_fs = 250;
+        } else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_500_DPS) {
+                gyr_fs = 500;
+        } else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_1000_DPS) {
+                gyr_fs = 1000;
+        } else if (fs_enum == GYRO_CONFIG0_GYRO_UI_FS_SEL_2000_DPS) {
+                gyr_fs = 2000;
+        }
+        return gyr_fs;
+}
+
+
+static int icm456xx_set_accel_power_mode(struct icm456xx_data *drv_data,
+					 const struct sensor_value *val)
+{
+	if ((val->val1 == ICM456XX_LOW_POWER_MODE) &&
+			(drv_data->accel_pwr_mode != ICM456XX_LOW_POWER_MODE)) {
+		if (drv_data->accel_hz != 0) {
+			if (drv_data->accel_hz <= 400) {
+				inv_imu_set_accel_mode(&drv_data->driver, PWR_MGMT0_ACCEL_MODE_LP);
+
+			} else {
+				LOG_ERR("Not supported ATTR value");
+				return -EINVAL;
+			}
+		}
+		drv_data->accel_pwr_mode = val->val1;
+	} else if ((val->val1 == ICM456XX_LOW_NOISE_MODE) &&
```

Comment:
Where is defined ICM456XX_LOW_NOISE_MODE? I assume there is a missing file to commit.

Reply:
I added ivm456xx.h file in /zephyr/include/zephyr/drivers/sensor/ directory.